ush/setup.py

```diff
@@ -1520,7 +1439,7 @@ def dict_find(user_dict, substring):
 
     # loop through the flattened expt_config and check validity of params
     cfg_v = load_config_file(os.path.join(USHdir, "valid_param_vals.yaml"))
-    for k, v in flatten_dict(expt_config).items():
+    for k, v in flatten_dict(var_defns_cfg).items():
```

Comment:
Flattening is the devil and should be removed ASAP.

Reply:
Structure is your friend.